assets/javascripts/discourse/components/new-topic-dropdown.js

```diff
@@ -22,6 +24,16 @@ export default DropdownSelectBoxComponent.extend({
     return this.dropdownButtons.buttons;
   },
 
+  get showButton() {
+    if (this.siteSettings.hide_new_topic_button_on_default_page) {
+      const currentRoute = this.router.currentRoute;
+      return currentRoute.name === `discovery.${defaultHomepage()}`
+        ? false
+        : true;
+    }
+    return true;
+  },
+
```

Comment:
We should move this to the connector.js, I'll how it would look like

Reply:
done in de367c1